src/app/content/highlights/reducer.ts

```diff
@@ -52,13 +54,31 @@ const reducer: Reducer<State, AnyAction> = (state = initialState, action) => {
         });
       }
 
+      const { locationFilterId, pageId } = action.meta;
+
+      const totalCountsPerPage = {...state.totalCountsPerPage};
+      if (totalCountsPerPage[pageId]) {
+        totalCountsPerPage[pageId] += 1;
+      } else {
+        totalCountsPerPage[pageId] = 1;
+      }
+
+      const totalCountsPerLocation = {...state.summary.totalCountsPerLocation};
```

Comment:
you wouldn't have to do this second part if you used a selector, its probably marginally more resource efficient to do it this way thought, simple addition rather than doing the whole object processing again. 

i could go either way, if you leave it like this please move some of this logic to a util function